cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -370,13 +375,64 @@ public void testSearchMetrics() throws Exception {
                                                       "system.writes", "system.zz", "user.reads", "user.writes"));
   }
 
+
   private void verifyAggregateQueryResult(String url, long expectedValue) throws Exception {
-    MetricQueryResult queryResult = post(url, MetricQueryResult.class);
+    // todo : can refactor this to test only the new tag name queries once we deprecate queryParam using context.
+    URI sourceUrl = new URI(url);
+    MetricQueryResult queryResult;
+    Map<String, List<String>> queryParams = new QueryStringDecoder(sourceUrl).getParameters();
+    if (!queryParams.containsKey("tag") || queryParams.get("groupBy").size() < 1) {
+      queryResult = post(url, MetricQueryResult.class);
+      Assert.assertEquals(expectedValue, queryResult.getSeries()[0].getData()[0].getValue());
+    }
+
+    // get new url replacing context with tags and groupBy (comma separated) into multiple query params
```

Comment:
we should do the other way - in test we use tag params, and here we also test old style api. Later we'd just remove old style extra testing without changing other code. Also that would make it less confusing for those who uses tests for reference on how to use rest pi